Python/errors.c

```diff
@@ -1850,6 +1850,52 @@ PyErr_SyntaxLocationEx(const char *filename, int lineno, int col_offset)
     Py_XDECREF(fileobj);
 }
 
+/* Raises a SyntaxError.
+ * If something goes wrong, a different exception may be raised.
+ */
+void
+_PyErr_RaiseSyntaxError(PyObject *msg, PyObject *filename, int lineno, int col_offset,
+                        int end_lineno, int end_col_offset)
+{
+    PyObject *text = PyErr_ProgramTextObject(filename, lineno);
+    if (text == NULL) {
+        text = Py_None;
```

Comment:
```suggestion
        text = Py_NewRef(Py_None);
```

Reply:
Yes.

Reply:
Oh, I see Serhiy's comment now.

Reply:
I proposed a solution to this question: https://github.com/python/cpython/issues/124074